website/docs/01-linters.md

```diff
@@ -0,0 +1,727 @@
+---
+id: linters
+title: "Infer : AL"
+---
+
+For C/C++ and Objective-C languages, we provide a linters framework. These are
+checks about the syntax of the program; it could be about a property, or about
+code inside one method, or that a class or method have certain properties. We
+provide [a few checks](/linters-bug-types) and we have developed a
+domain specific language (DSL) to make it easier to write checks.
+
+- [AL: A declarative language for writing linters in Infer](/linters#al_intro)
+- [Getting the clang AST](/linters#clang_ast)
+- [Using AL to write linters](/linters#write_linters)
+  - [General structure of a checker](/linters#checker_structure)
+  - [Defining Paths](/linters#paths)
+  - [Defining Macros](/linters#macros)
+  - [AL Predicates](/linters#predicates)
+  - [AL Formulas](/linters#formulas)
+  - [Example AL checkers:](/linters#examples)
+    - [Objective-C class that inherits from a class that shouldn't be subclassed](/linters#subclass_check)
+    - [Objective-C instance method call](/linters#instance_method_check)
+    - [Objective-C instance method call of any method of a class](/linters#any_instance_method_check)
+    - [Objective-C class method call](/linters#class_method_check)
+    - [Objective-C method call of a method with int return type](/linters#return_type)
+    - [Variable declaration with type long](/linters#variable_type)
+    - [Method that has a parameter of type A\*](/linters#some_parameter)
+    - [Method that has all the parameters of type A\* (and at least one)](/linters#all_parameters)
+    - [Method that has the 2nd parameter of type A\* ](/linters#sec_parameter)
+    - [Protocol that inherits from a given protocol](/linters#protocol)
+    - [Constructor is defined with a parameter of a type that implements a given protocol](/linters#constructor_protocol_type)
+    - [Variable declaration of type NSArray applied to A](/linters#nsarray_generics)
+    - [Using a property or variable that is not available in the supported API](/linters#unavailable_api)
+    - [Using a given namespace](/linters#use_namespace)
+    - [A check for flagging the use of given enum constants](/linters#enum_constants)
+  - [AST info in messages](/linters#info_message)
+  - [Testing your rule](/linters#testing)
+  - [Debugging](/linters#debugging)
+  - [Demo](/linters#demo)
+  - [Command line options for linters](/linters#command_line)
```

Comment:
Dead links (missing "docs/"?).

The sections in this document are written with `**` instead of `##`, unfortunately, otherwise we could just delete this list and rely on the local navigation sidebar. (making a note to fix that later)

Reply:
Fixed